Some UI updates and features

verdantPermission opened the pull request:
- add ability to change player names in-place
- modify scorecard layout to occupy full space regardless of the number of players
- store the application state in localStorage to preserve names, # player selection, language, even scores

verdantPermission committed:
- add ability to change player names in-place
- modify scorecard layout to occupy full space regardless of the number of players
- store the application state in localStorage to preserve names, # player selection, language, even scores

greengiraffe commented:
@verdantPermission Thanks! You basically implemented all the nice-to-have features that I had in my backlog (aka my head), so this is very much appreciated. :)

One minor thing: the German translation for the player reset button causes a line break, and the button's padding is not pixel perfect (yes, that's nitpicking). I'll merge it for now but add an issue for that. EDIT: Issue backlink: https://github.com/greengiraffe/wingspan-helper/issues/11

Also, maybe I should remove the Automata part as you did... I don't find the time for it at the moment.

greengiraffe committed:
:page_facing_up: Add @verdantPermission to LICENSE

verdantPermission commented:
Cool, thanks for accepting my PR. I saw your enhancement Issues only after I had finished my changes :-)
I wasn't super-happy with the look of the player reset button, but got tired of CSS tweaking. I'm not completely convinced that it's obvious what that "Reset" button does, but I wanted a way to clear the names, and that seemed the logical place to put a button. I also wanted to keep that column's width narrow to allow maximum space for names, so I had competing concerns. Maybe "Reset Names" with a line-break, and smaller font? Anyways, my primary use-case is using the installed web-app on my phone, and the important thing is when I win a game against my girlfriend, my screenshot has our names :-)

greengiraffe commented:
That's a very good use-case! :-)

I decided to make the player name reset button more subtle and only show it when a name has been changed. This should make it obvious what its used for. Also, I released [v.1.1.0](https://github.com/greengiraffe/wingspan-helper/releases/tag/v1.1.0) that includes your changes, the updated button style, underlining of winning scores, as well as a new _Nectar_ row for the Oceania expansion which I received a few days ago. ;-)

Hope you like it!

verdantPermission commented:
I'd heard about Nectar in the Oceania expansion (I'm waiting for my order!)
I like the player reset button only appearing after the names have been
changed.
I tried the new version, but I think there's something wrong with the
winning score underlining: it's showing more than one total underlined
(different scores).

On Mon, Dec 21, 2020 at 4:08 PM Valentin von Guttenberg <
notifications@github.com> wrote:

> That's a very good use-case! :-)
>
> I decided to make the player name reset button more subtle and only show
> it when a name has been changed. This should make it obvious what its used
> for. Also, I released v.1.1.0
> <https://github.com/greengiraffe/wingspan-helper/releases/tag/v1.1.0>
> that includes your changes, the updated button style, underlining of
> winning scores, as well as a new *Nectar* row for the Oceania expansion
> which I received a few days ago. ;-)
>
> Hope you like it!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/greengiraffe/wingspan-helper/pull/10#issuecomment-749196586>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARGAKDXMVDFDN477SK2BEXDSV62N3ANCNFSM4URLZ42Q>
> .
>

greengiraffe commented:
That's odd. It works for me in Firefox and Chrome. What browser are you using?

verdantPermission commented:
Using Chrome, I cleared my cache, deleted my stored state.
Started with 4,5,6,7,8 in the birds row, and showed Total, and All the
totals were underlined.
Added some bonus scores, and getting odd behaviour.

[image: image.png]


On Mon, Dec 21, 2020 at 5:09 PM Valentin von Guttenberg <
notifications@github.com> wrote:

> That's odd. It works for me in Firefox and Chrome. What browser are you
> using?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/greengiraffe/wingspan-helper/pull/10#issuecomment-749224391>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARGAKDUOUNZW2PFNCBXYUSTSV7BSPANCNFSM4URLZ42Q>
> .
>

greengiraffe commented:
Thanks, I could reproduce that. Will fix it tomorrow.

greengiraffe commented:
Fixed. Used a wrong compare function for sorting the scores.

verdantPermission commented:
Fix confirmed. Looks great!
I only have a few tiny suggestions for 'improvement', take them as you wish:

   - for the new Nectar row, would be nice to somehow indicate it only
   applies to the Oceania expansion. I don't think the row should show/hide,
   just if someone only has the base game, they might wonder "What's Nectar?
   What am I missing?"
   - in my fork, I reduced the left/right margins and padding as much as
   possible, and squeezed the left column slightly to give maximum space for
   the names. With 5 players, on a phone screen, it's a little cramped
   - on a phone-width screen (I use Chome's DevTools device toolbar, my
   phone display is closest to 'Pixel 2' size), on the bottom line, there's an
   unfortunate line-break in the middle of "Nils Cordes"

[image: image.png]


On Tue, Dec 22, 2020 at 11:45 AM Valentin von Guttenberg <
notifications@github.com> wrote:

> Fixed. Used a wrong compare function for sorting the scores.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/greengiraffe/wingspan-helper/pull/10#issuecomment-749648422>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARGAKDQ7DFOKFZLH5NYH7PLSWDEI7ANCNFSM4URLZ42Q>
> .
>

greengiraffe commented:
Yes, great suggestions! I've channeled them into issue #13. Feel free to create a PR to include your changes on margins/paddings and font size.
